app/lib/dartdoc/models.dart

```diff
@@ -84,30 +83,30 @@ class DartdocEntry extends Object with _$DartdocEntrySerializerMixin {
   List<int> asBytes() => UTF8.encode(JSON.encode(this.toJson()));
 
   TaskTargetStatus checkTargetStatus({bool retryFailed: false}) {
-    if (isNewer(_semanticDartdocVersion, versions.semanticDartdocVersion)) {
-      // our dartdoc version is newer -> regenerate
-      return new TaskTargetStatus.ok();
-    }
-    if (isNewer(versions.semanticDartdocVersion, _semanticDartdocVersion)) {
-      // our dartdoc version is older -> skip generating
-      return new TaskTargetStatus.skip('Entry has newer dartdoc version.');
-    }
-
-    if (usesFlutter &&
-        isNewer(_semanticFlutterVersion, versions.semanticFlutterVersion)) {
-      // our flutter version is newer -> regenerate
+    final age = new DateTime.now().difference(timestamp).abs();
+    if (age > entryUpdateThreshold) {
       return new TaskTargetStatus.ok();
     }
 
-    if (isNewer(
-        _semanticCustomizationVersion, versions.semanticCustomizationVersion)) {
-      // our customization version is newer -> regenerate
+    if (runtimeVersion == null) {
+      // legacy data -> regenerate
```

Comment:
Could we somehow skip the regeneration and just update the datastore entry, if hasContent and the versions are the same as current?

Reply:
Good point, but instead of checking each version, I've opted for a hardcoded `?? '2018.3.8'` for now. PTAL.